tree/tree/inc/TTree.h

```diff
@@ -160,6 +163,11 @@ class TTree : public TNamed, public TAttLine, public TAttFill, public TAttMarker
    TTree(const TTree& tt);              // not implemented
    TTree& operator=(const TTree& tt);   // not implemented
 
+#ifdef R__USE_IMT
+   mutable Bool_t fIMTFlush{false};               ///<! True if we are doing a multithreaded flush.
```

Comment:
Yes, that's the intention.  Well, the real problem here is that `FlushBaskets` is declared `const`: the `mutable` here is working around that.

The `fIMTFlush` is only called from within `FlushBaskets`, which is not allowed to be called from multiple threads.  It's done outside the parallel portion, meaning there must be a synchronization after it is mutated.

I wanted to avoid making it atomic to fulfill the request that there is no performance penalty when ROOT is used in non-IMT mode.